### PR TITLE
Improve folder management

### LINKED ISSUE #32: Improve ergonomics & feedback
## Problems

1. User must run `new` then `cd` the new folder and then run `start-env`. Failure to do so will return unclear errors.
2. If a user tries to run `start-env` twice, he gets unclear errors.
3. The folders used to mount docker volumes, currently stored at `/tmp/create-comit-app` are not cleaned-up.

## Goal

1. Remove the `cd` step.
2. Explicitly stop the user to run `start-env` twice on the same machine.
3. Clean-up the folders used for volume
4. Update hello-swap if needed (open new issue)

## Recommendation

As per the [Linux Information Project](http://www.linfo.org/var.html):

> Among the various subdirectories within /var are /var/cache (contains cached data from application programs), /var/games (contains variable data relating to games in /usr), /var/lib (contains dynamic data libraries and files), /var/lock (contains lock files created by programs to indicate that they are using a particular file or device), /var/log (contains log files), /var/run (contains PIDs and other system information that is valid until the system is booted again) and /var/spool (contains mail, news and printer queues).

1. Hence, the proposal would be to replace `./.env` with `/var/run/create-comit-app/env`.

2. This could then be used as a `lock` file, meaning, if the file exists and a user runs `start-env` then an error should explain that it seems like an instance is already running:

```
▶ create-comit-app start-env > /dev/null 2>&1 &
[1] 71840
▶ create-comit-app start-env
An environment is already running!
If you are sure that no other `create-comit-app start-env` process is running, delete the lock file at `/var/run/create-comit-app/env`.
```

3. Finally,  move the temp folders under the same `/var/run/create-comit-app/` so start-env can delete the whole folder at the end (instead of just the env file).

### PR DESCRIPTION
**Pull request by D4nte**
Resolves #32.

### CODE REVIEW

File: src/start_env/mod.rs [hunk 1]
```diff
@@ -25,7 +25,7 @@ use tokio::runtime::Runtime;
 use tokio::timer::Interval;
 use web3::types::U256;
 
-const ENV_FILE_PATH: &str = ".env";
+mod path;
```
[the remark this answers is not in the document]
Comment (luckysori) [reply src/start_env/mod.rs]: Actually, I guess `path` is doing more than environment variable stuff! I still don't like the name though. More like `temp_folder`?

File: src/start_env/mod.rs [hunk 2]
```diff
@@ -369,10 +375,15 @@ fn start_cnds(envfile_path: &PathBuf) -> impl Future<Item = Vec<Node<Cnd>>, Erro
 }
 
 fn temp_folder() -> PathBuf {
```
Comment (luckysori) [inline src/start_env/mod.rs]: :question: Couldn't this folder just live inside `path`?

File: src/start_env/mod.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (D4nte) [reply src/start_env/mod.rs]: `temp_fs` ?

File: src/start_env/mod.rs [hunk 3]
```diff
@@ -118,26 +123,25 @@ fn start_all() -> Result<Services, Error> {
         })
         .collect::<Vec<_>>();
 
-    let envfile_path = PathBuf::from(ENV_FILE_PATH);
-    std::fs::File::create(envfile_path.clone())
-        .unwrap_or_else(|_| panic!("Could not create {} file", ENV_FILE_PATH));
+    let env_file_path = path::env_file_path();
+    path::create_env_file().unwrap_or_else(|_| panic!("Could not create {:?} file", env_file_path));
```
[the remark this answers is not in the document]
Comment (D4nte) [reply src/start_env/mod.rs]: Let's merge that and fix error handling in the PR I working on for #38.

File: src/start_env/mod.rs [hunk 4]
```diff
@@ -192,7 +197,8 @@ fn start_all() -> Result<Services, Error> {
     envfile.write().map_err(|e| {
         eprintln!(
             "Could not write {} file, aborting...\n{:?}",
-            ENV_FILE_PATH, e
+            path::env_file_str(),
```
[the remark this answers is not in the document]
Comment (D4nte) [reply src/start_env/mod.rs]: yes but you get a result. Let's fix all error handling in my current PR for #38.

File: src/start_env/mod.rs (hunk 2 above)
Comment (D4nte) [reply src/start_env/mod.rs]: Yep, sounds good, I'll move it in my current PR :)

File: CHANGELOG.md [hunk 5]
```diff
@@ -4,6 +4,15 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+## Added
+- Ensure that all temporary folder are cleaned up when shutting down start-env.
```
Comment (luckysori) [inline CHANGELOG.md]: :wrench: Typo.
```suggestion
- Ensure that all temporary folders are cleaned up when shutting down start-env.
```

**Pull request merged**